server/auth/permission.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissionKeys and
+// limitations under the License.
+//
+
+package auth
+
+import (
+	"strings"
+
+	"github.com/tikv/pd/pkg/errs"
+)
+
+// Action represents rbac actions.
+type Action string
+
+// All available actions types.
+const (
+	GET    Action = "get"
+	LIST   Action = "list"
+	CREATE Action = "create"
+	UPDATE Action = "update"
+	DELETE Action = "delete"
+)
+
+// Permission represents a permission to a specific pair of resource and action.
+type Permission struct {
+	Resource string `json:"resource"`
```

Comment:
Should this be a newtype like Action as well?

Reply:
We may define `Resource` as an enum.